sdk/src/wallet/operations/balance.rs

```diff
@@ -121,11 +141,23 @@ where
                         // add nft_id for nft outputs
                         if let Output::Nft(nft) = &output {
                             let nft_id = nft.nft_id_non_null(output_id);
-                            balance.nfts.push(nft_id);
+                            balance.nfts.insert(nft_id);
                         }
 
                         // Add amount
                         balance.base_coin.total += output.amount();
+                        // Add stored mana
+                        balance.mana.total.stored_mana += protocol_parameters.mana_with_decay(
+                            output.mana(),
+                            output_id.transaction_id().slot_index(),
+                            slot_index,
+                        )?;
+                        // Add potential mana
+                        balance.mana.total.potential_mana += protocol_parameters.generate_mana_with_decay(
+                            output.amount(),
```

Comment:
The required amount for the storage deposit needs to be subtracted here and in the other places

Reply:
Are you sure? Isn't that still generating mana for this wallet?

Reply:
Yes `Every IOTA coin (besides storage deposits) generates potential Mana.` https://github.com/iotaledger/tips/blob/tip39/tips/TIP-0039/tip-0039.md#potential-mana

Reply:
ahh